### PR TITLE
cleanup cluster object

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Cleanup cluster object:
- Remove fields which are not in use since a long time
- Mark deprecated fields

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/cmd/kubermatic-api/swagger.json
Comment (p0lyn0mial) [inline api/cmd/kubermatic-api/swagger.json:2666]: you changed the internal types but this was reflected in the swagger spec :)

**Comment by p0lyn0mial** [comment]
@mrIncompetent are those changes backward compatible ?
@mrIncompetent what about already existing resources ?

File: api/cmd/kubermatic-api/swagger.json
Comment (mrIncompetent) [reply api/cmd/kubermatic-api/swagger.json:2666]: Yeah, as we still pass through the full cluster object in the old endpoints

**Comment by mrIncompetent** [comment]
> @mrIncompetent are those changes backward compatible ?
@mrIncompetent what about already existing resources ?

No running kubermatic installation relies on those fields anymore.

**Comment by p0lyn0mial** [comment]
this LGTM if the removed fields are not used by the client, that is 

`kubeletToken`, `workerName` and `lastDeployedMasterVersion`

I'm almost sure they are not, is that correct ?

**Comment by mrIncompetent** [comment]
> this LGTM if the removed fields are not used by the client, that is
> kubeletToken, workerName and lastDeployedMasterVersion
> I'm almost sure they are not, is that correct ?

I just searched for usages in the dashboard-v2:
Only in fake objects. And those do not need to be updated now. As we're migrating to the new endpoints, those fields will be deleted anyway

**Comment by p0lyn0mial** [comment]
> As we're migrating to the new endpoints, those fields will be deleted anyway

I know that they will be deleted. I just want to make sure that those changes will not break the dashboard and the users will be able to create `cluster` resources.

I'd set up a local environment and perform a manual test but that's your call.

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**
